mwzeval/metrics.py

```diff
@@ -306,10 +306,10 @@ def block_domains(input_states: dict, reference_states: dict, blocked_domains: s
                 # drop the blocked slots from the reference state
                 new_turn_ref = {}
                 for domain, slot_values in turn_ref.items():
-                    if domain not in blocked_domains:
+                    if domain in included_domains:
                         new_turn_ref[domain] = slot_values
 
-                # if the reference state does not contain any unblocked slot,
+                # if the reference state does not contain any slot from the included domain,
```

Comment:
What if you have a beserk model that predicts states in the wrong domain when it should not? With the current code, the model would get away with such errors. Would it not make more sense to drop the turn if the ref does not contain that domain and the model also did not predict it?